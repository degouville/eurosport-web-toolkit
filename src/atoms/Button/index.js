import React from 'react';
import styled from 'styled-components';

const StyledButton = styled.button`
  background-color: red;
  color: yellow;
`;

export default ({ children }) => <StyledButton type="button">{children}</StyledButton>;
