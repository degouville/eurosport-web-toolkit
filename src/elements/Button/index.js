import styled, { css } from 'react-emotion';
import { lighten, rgba } from 'polished';
import PropTypes from 'prop-types';
import ArrowLink from '../ArrowLink';

const Button = styled(ArrowLink)`
  background-color: ${({ theme, type }) => theme.button[type].backgroundColor};
  color: ${({ theme, type }) => theme.button[type].color};
  cursor: pointer;
  padding: 18px 10px;
  border-radius: 3px;
  font-size: 12px;
  letter-spacing: 2px;
  line-height: 14px;

  &:focus,
  &:visited,
  &:hover {
    color: ${({ theme, type }) => theme.button[type].color};
  }

  &:after {
    margin-left: 11px;
  }

  ${({ theme, type }) =>
    type === 'primary' &&
    css`
      &:hover {
        background-color: ${lighten(0.2, theme.button.primary.backgroundColor)};
      }
    `};

  ${({ theme, type }) =>
    type === 'secondary' &&
    css`
      border: 1px solid ${rgba(theme.button.secondary.color, 0.4)};
      &:hover {
        border-color: ${theme.button.secondary.color};
      }
    `};

  ${({ theme, type }) =>
    (type === 'form' || type === 'secondaryForm') &&
    css`
      &:hover {
        background-color: ${lighten(0.05, theme.button[type].backgroundColor)};
      }
      transition: all 150ms ease;
    `};
`;

Button.defaultProps = {
  type: 'primary',
};

Button.propTypes = {
  type: PropTypes.oneOf(['primary', 'secondary', 'form', 'secondaryForm']),
};

Button.displayName = 'Button';

export default Button;
