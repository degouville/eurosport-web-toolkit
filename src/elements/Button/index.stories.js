import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import { withInfo } from '@storybook/addon-info';
import { Button } from '../..';

const indexStories = storiesOf('Button', module);

indexStories
  .add('primary', withInfo()(() => <Button type="primary">{text('Children', 'Children prop text')}</Button>))
  .add('secondary', withInfo()(() => <Button type="secondary">{text('Children', 'Children prop text')}</Button>));
