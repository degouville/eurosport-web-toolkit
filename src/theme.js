import { transparentize, rgba } from 'polished';
import {
  plausibleViolet,
  crashingAmber,
  coreLightMinus1,
  significantAzure,
  featureOneMinus2,
  actionTwoDarkBase,
  coreDarkPlus1,
  coreNeutral2,
  coreNeutral4,
  coreNeutral8,
  coreLightBase,
  dodgerBlue2,
  actionTwoLightBase,
  melanzane,
  coreDarkBase,
  hydroelectricTar,
  coreNeutral6,
  coreNeutral1,
  actionOneDarkMinus1,
  actionOneDarkBase,
  ebonyClay,
  martinique,
  suitGray,
  blackPearl,
} from './colors';
import { fontAlphaHeadline, fontInterUi } from './typography';

export default {
  underlineLink: {
    fontFamily: fontInterUi,
    color: coreNeutral4,
    hoverColor: coreNeutral2,
  },
  link: {
    color: featureOneMinus2,
  },
  button: {
    primary: {
      color: coreLightMinus1,
      backgroundColor: actionTwoDarkBase,
    },
    secondary: {
      color: featureOneMinus2,
    },
    form: {
      color: coreLightMinus1,
      backgroundColor: dodgerBlue2,
    },
  },
  input: {
    backgroundColor: coreDarkPlus1,
    borderFocused: coreNeutral4,
    border: coreNeutral8,
    placeholder: {
      color: coreNeutral4,
      fontFamily: fontInterUi,
    },
    text: {
      color: coreLightBase,
      fontFamily: fontInterUi,
    },
  },
  login: {
    separator: {
      primary: coreLightBase,
      secondary: coreNeutral8,
    },
    text: {
      color: coreLightBase,
      fontFamily: fontInterUi,
    },
    title: {
      color: coreLightMinus1,
      fontFamily: fontAlphaHeadline,
    },
  },
  banner: {
    error: {
      borderColor: actionTwoLightBase,
      backgroundColor: melanzane,
      textColor: coreLightBase,
      fontFamily: fontInterUi,
    },
  },
  tab: {
    active: {
      color: coreLightMinus1,
      borderColor: featureOneMinus2,
    },
    borderColor: transparentize(0.5, coreLightMinus1),
  },
  typo: {
    primary: {
      color: coreLightMinus1,
    },
  },
  playerControls: {
    dropdown: {
      background: transparentize(0.1, coreDarkBase),
      border: transparentize(0.2, coreNeutral6),
    },
    seek: {
      trackColor: dodgerBlue2,
      handleSize: 20,
      handleColor: coreLightBase,
      railColor: rgba(coreNeutral1, 0.3),
      railThickness: 5,
      vertical: false,
    },
    volume: {
      trackColor: dodgerBlue2,
      handleSize: 20,
      handleColor: coreLightBase,
      railColor: rgba(coreNeutral1, 0.3),
      railThickness: 5,
      vertical: true,
    },
    bar: {
      backgroundColor: transparentize(0.4, coreDarkBase),
    },
    separator: transparentize(0.5, coreNeutral4),
    liveIcon: {
      fontFamily: fontAlphaHeadline,
      backgroundColor: actionTwoDarkBase,
      color: coreLightBase,
    },
    text: {
      fontFamily: fontAlphaHeadline,
      color: coreLightBase,
    },
  },
  dropdown: {
    label: {
      color: coreLightMinus1,
    },
    list: {
      default: {
        color: coreNeutral2,
      },
      selected: {
        color: coreLightMinus1,
      },
    },
    checkMark: {
      color: significantAzure,
    },
    background: hydroelectricTar,
    border: transparentize(0.2, coreNeutral6),
  },
  scrollbar: {
    color: plausibleViolet,
    background: crashingAmber,
  },
  statistics: {
    comparison: {
      colorOne: actionOneDarkMinus1,
      colorTwo: actionOneDarkBase,
      background: rgba(coreLightMinus1, 0.1),
    },
    background: {
      one: ebonyClay,
      two: martinique,
    },
    fringe: rgba(coreLightMinus1, 0.05),
  },
  marketingChecklist: {
    title: {
      color: coreLightMinus1,
    },
    subtitle: {
      color: suitGray,
    },
    check: {
      color: coreLightMinus1,
    },
  },
  videoPlayerModal: {
    closeIcon: {
      fill: coreLightMinus1,
    },
  },
  loginWithMarketing: {
    mobileBackground: blackPearl,
  },
};
