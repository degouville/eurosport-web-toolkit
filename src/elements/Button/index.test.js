import React from 'react';
import { shallow } from 'enzyme';
import theme from '../../theme';
import Button from '.';

it('renders a Button', () => {
  expect(shallow(<Button theme={theme}>Click me !</Button>)).toMatchSnapshot();
});
