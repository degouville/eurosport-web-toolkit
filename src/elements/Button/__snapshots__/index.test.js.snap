// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a form Button 1`] = `
<Button
  type="form"
>
  Click me !
</Button>
`;

exports[`Button renders a primary Button by default 1`] = `
<Button
  type="primary"
>
  Click me !
</Button>
`;

exports[`Button renders a secondary Button 1`] = `
<Button
  type="secondary"
>
  Click me !
</Button>
`;
